php/Terminus/Commands/CommandWithSSH.php

```diff
@@ -120,6 +120,24 @@ protected function getAppserverInfo(array $site_info = array()) {
   }
 
   /**
+   * Checks the site's mode and suggests SFTP if it is not set.
+   *
+   * @param Site $site Site object to check the connection mode for
+   * @return void
+   */
+  protected function checkConnectionMode($site) {
+    $mode = $site->environments->get('dev')->info('connection_mode');
+    if ($mode != 'sftp') {
+      $message  = 'Change your connection mode to SFTP to facilitate ';
+      $message .= 'file changes by made by {client}.';
```

Comment:
A couple changes:

1. This should check the environment targeted by the command, and only care if it's not test or live. E.g. if I'm running drush against test, there is no SFTP mode. Likewise, if I am targeting a multidev env (feature-x) that's where we should check, not the dev environment. 
2. The message is a little vague. I would suggest something more like: `Note: this environment is in read-only Git mode. If you need to make changes to the codebase of this site (e.g. updating modules or plugins) you will need to toggle into read-write SFTP mode first.`